src/server/authserver/Server/AuthSocket.cpp

```diff
@@ -208,6 +208,10 @@ void AuthSocket::OnRead()
 {
     #define MAX_AUTH_LOGON_CHALLENGES_IN_A_ROW 3
     uint32 challengesInARow = 0;
+
+    #define MAX_AUTH_GET_REALM_LIST 10
```

Comment:
we could make it configurable via `authserver.conf`

Reply:
Good idea, i think

Reply:
This is a critical bug, I think that the priority is test and merge this PR asap.

I will add the config when meanwhile I'll make the PR for this issue https://github.com/azerothcore/azerothcore-wotlk/issues/2383